### ISSUE TITLE
Test compatibility of the jQuery delivered in oik with WordPress 5.5 and jQuery v3.5.1 

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/oik#153]
- WordPress 5.5, delivered on 11th August 2020, removed the automatic enqueing of jQuery-migrate.
- In a future version of WordPress, jQuery will be updated to v3.5.1.
- The oik shortcodes deliver a number of jQuery files.
- These are used by shortcodes such as `[bw_countdown]` and `[bw_cycle]` and a general purpose shortcode to run jQuery  `[bw_jq]`.
- These `jquery.*.js` files were last updated in 2016
- There are other shortcodes that enqueue JavaScript files; these should also be tested
- And there are other .js files that should be tested.

### Requirements

- [x] Test that these work without the need for jQuery-migrate
- [x] Test the behaviour with jQuery 3.5.1
- [x] With/without the new jQuery migrate
- [x] If there's anything wrong decide what to do
- [x] Such as updating the files to later versions
- [x] Or deprecating the logic
- [x] Update any PHPUnit tests, as required

The list of shortcodes to test includes:
- [x] bw_accordion - uses `jquery-ui-accordion`
- [x] bw_countdown - uses `countdown`
- [x]  bw_cycle - uses `cycle.all`
- [x] bw_jq - uses whatever the user chooses
- [x] bw_logo - uses - oik_bw_logo.js
- [x] bw_tabs - uses `jquery-ui-tabs`
- [x] bw_testimonials - uses `cycle.all`

Other jQuery to test includes:
- [x] oik-quicktags
- [x] bw_shortcodes
- [x] oik_shortc_plugin (_4)
- [x] oik_button_plugin
- [x] oik_paypal_plugin.js

Dependent plugins to test include:

- [x] oik-css
- [x] oik-nivo-slider - see bobbingwide/oik-nivo-slider/issues/11
- [x] oik-popup - uses `fancybox` - see #154 
- [x] oik-squeeze - uses fancybox for the squeeze popup
- [x] oik-external-link-warning - `jquery-ui-tooltop` - see https://github.com/bobbingwide/oik-external-link-warning/issues/4

### Method
- Test on a local copy of oik-plugins.com
- Use a combination of WordPress, Gutenberg and WordPress SEO that is believed to be stable; see below
- Test with `wp-jquery-update-test` plugin with the 8 different combinations of radio buttons selected. 
- Report any issues here, creating new issues for tracking and version control.

Combinations which are believed to be stable are:

WordPress |  Gutenberg | WordPress SEO 
------ | ------ | --------
5.5 | 8.8.0 | 14.8
5.5 + fix for TRAC 50999 | 8.7.1 | 14.7 
5.5 | 8.8.0 | 14.8.1 
5.5.1 | 8.8.0 | 14.9

**Comment by bobbingwide** [comment]
Bugger. I lost another comment which had a lot of good stuff.

**Comment by bobbingwide** [comment]
jQuery Fancybox needs jQuery Migrate. 

It doesn't work at all with jQuery 3.5.1

```
jquery.fancybox-1.3.4.pack.js?ver=5.5:18 Uncaught TypeError: Cannot read property 'msie' of undefined
    at jquery.fancybox-1.3.4.pack.js?ver=5.5:18
    at jquery.fancybox-1.3.4.pack.js?ver=5.5:46
(anonymous) @ jquery.fancybox-1.3.4.pack.js?ver=5.5:18
(anonymous) @ jquery.fancybox-1.3.4.pack.js?ver=5.5:46
(index):361 WordPress jQuery: 1.12.4, Migrate: not available, UI: 1.11.4
(index):369 Uncaught TypeError: jQuery(...).fancybox is not a function
    at HTMLDocument.<anonymous> ((index):369)
    at i (jquery.js?ver=1.12.4-wp:2)
    at Object.fireWith [as resolveWith] (jquery.js?ver=1.12.4-wp:2)
    at Function.ready (jquery.js?ver=1.12.4-wp:2)
    at HTMLDocument.J (jquery.js?ver=1.12.4-wp:2)
```

**Comment by bobbingwide** [comment]
On retesting I found that the [bw_testimonials] worked for all combinations but the [bw_cycle] examples were badly styled when jQuery version was 3.5.1
Also, for none of the 8 combinations did I get anything other than `UI: not available`

**Comment by bobbingwide** [comment]
I noted that the documentation for `[bw_jq]` was rather lacking. And the syntax didn't reflect the order of the positional parameters. So update the code and PHPUnit tests.

**Comment by bobbingwide** [comment]
I haven’t checked all uses of the following APIs in plugins dependent upon oik or oik shared libraries.
- bw_jquery
- bw_jquery_enqueue_script
- bw_jquery_enqueue_style

**Comment by bobbingwide** [comment]
At the end of August I'd created some examples for `[bw_jq]` with `pullquote` and they worked. When I retested them they didn't work until I ensured that the `<span class="pq">` was nested in a paragraph tag `<p>`. I then realised that the example was sensitive to the value of the **Disable automatic paragraph creation** checkbox. So I changed the example so that it wasn't, and wrote some notes.   

PS. The pullquote jQuery is not something I'd expect anyone to use unless they really wanted to.
